src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,218 @@
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT c.id AS car_id, c.model, m.id AS manufacturer_id, m.name "
+                      + "AS manufacturer_name, m.country FROM cars c "
+                      + "JOIN manufacturers m ON c.manufacturer_id = m.id "
+                      + "WHERE c.id = ? AND c.is_deleted = FALSE;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get car by id " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getAllDrivers(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT c.id AS car_id, c.model, m.id AS manufacturer_id, m.name "
+                + "AS manufacturer_name, m.country FROM cars c JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.id WHERE c.is_deleted = FALSE "
+                + "AND m.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get list of cars from cars table", e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getAllDrivers(car.getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                     + "WHERE id = ? AND is_deleted = FALSE;";
+        int result = 0;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.setLong(3, car.getId());
+            result = preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car " + car, e);
+        }
+        if (result > 0) {
+            removeDriverFromCar(car.getId());
+            addDriverToCar(car);
+            return car;
+        }
+        throw new RuntimeException("There is nothing to update");
```

Comment:
create constructor without exception and throw DataProcessingExc instead of Runtime

Reply:
done